packages/mermaid/src/mermaid.ts

```diff
@@ -270,6 +270,23 @@ if (typeof document !== 'undefined') {
    * Wait for document loaded before starting the execution
    */
   window.addEventListener('load', contentLoaded, false);
+  window.addEventListener('error', ({ message }) => {
+    // Get mermaid function called from error message with regex
+    const match = message.match(/mermaid\.(\w+) is not a function/);
+    if (match) {
+      const notify = () =>
+        // eslint-disable-next-line no-console
+        console.error(`------------------------------
+Breaking change in Mermaid v11
+------------------------------
+Use mermaid.default.${match[1]}() instead of mermaid.${match[1]}()
+Read more: https://github.com/mermaid-js/mermaid/releases/tag/v11.0.0
+`);
+      notify();
+      // Log again after a delay to ensure it's the last message
+      setTimeout(notify, 1000);
+    }
+  });
```

Comment:
> There's got to be a way to export everything as globalThis.mermaid instead of globalThis.mermaid.default.

I couldn't find any, that's why this suboptimal method was chosen until we could have some fix. 
Let me try the footer option.

Reply:
That worked! 🚀

Reply:
If that is also the same, is this really a breaking change? 😅 
What does it break from a user's perspective?

Reply:
> What does it break from a user's perspective?

UMD supports CommonJS, aka doing `const mermaid = require("mermaid/dist/mermaid.min.js");` IIFE does not (although something like `require("mermaid/dist/mermaid.min.js"); const mermaid = globalThis.mermaid;` might still work?).

Also, UMD supports [AMD](https://github.com/amdjs/amdjs-api/blob/master/AMD.md), which is a mysterious JavaScript module format used by the ancient JavaScript wizards :mage:, many eons ago, back when dinosaurs :sauropod: ruled the world [^1].

Personally, I highly doubt anybody is using CJS/AMD for mermaid, but just in case, we need to make this a breaking change.

[^1]: By dinosaurs, I mean Internet Explorer, which was still the most popular browser in 2011 (when AMD was being used) https://gs.statcounter.com/#desktop-browser-ww-monthly-201107-201107-bar